packages/gatsby-recipes/src/renderer/render.js

```diff
@@ -253,48 +257,53 @@ const render = (recipe, cb, inputs = {}, isApply, isStream, name) => {
   const renderResources = isDrained => {
     result = RecipesReconciler.render(recipeWithWrapper, plan, name)
 
-    const resources = transformToPlan(result)
+    const resourcesArray = transformToPlan(result)
 
     const isDone = () => {
+      let result
       // Mostly for validation stage that checks that there's no resources
       // in the initial step — this done condition says no resources were found
       // and there's no inflight resource work (resources will be empty until the
       // first resource returns).
       //
       // We use "inFlightCache" because the queue doesn't immediately show up
       // as having things in it.
-      if (resources.length === 0 && ![...inFlightCache.values()].some(a => a)) {
-        return true
+      if (
+        resourcesArray.length === 0 &&
+        ![...inFlightCache.values()].some(a => a)
+      ) {
+        result = true
         // If there's still nothing on the queue and we've drained the queue, that means we're done.
       } else if (isDrained && queue.length === 0) {
-        return true
+        result = true
         // If there's one resource & it fails validation, it doesn't go into the queue
         // so we check if inFlightCache is empty & all resources have an error.
       } else if (
-        !resources.some(r => !r.error) &&
+        !resourcesArray.some(r => !r.error) &&
         ![...inFlightCache.values()].some(a => a)
       ) {
-        return true
+        result = true
+      } else {
+        result = false
       }
 
-      return false
+      return result
     }
 
     if (isDone()) {
       // Rerender with the resources and resolve the data from the cache.
-      emitter.emit(`done`, resources)
+      emitter.emit(`done`, resourcesArray)
     }
   }
 
-  const throttledRenderResources = lodash.throttle(renderResources, 30, {
+  const throttledRenderResources = lodash.throttle(renderResources, 100, {
     trailing: false,
   })
 
   queue.on(`task_finish`, function (taskId, r, stats) {
     throttledRenderResources()
 
-    const resources = transformToPlan(result)
-    emitter.emit(`update`, resources)
```

Comment:
Tricky. Before it was guaranteed to run the `transformToPlan` prior to emitting the `update` event. Afterwards that's not the case and the information in `resourcesArray` may be stale?

Should the `emit` line be at the end of the `renderResources` function instead?

Reply:
hmm true, lemme think about this

Reply:
ok solution, just move emitting updates to run after `transformToPlan`